internal/authctx/selfmanaged.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package authctx
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+	oidcapi "go.pinniped.dev/generated/latest/apis/supervisor/oidc"
+	"go.pinniped.dev/pkg/oidcclient/pkce"
+	"go.pinniped.dev/pkg/oidcclient/state"
+	"golang.org/x/oauth2"
+)
+
+const (
+	tokenEndpointSuffix         = "oauth2/token"
+	authorizationEndpointSuffix = "oauth2/authorize"
+	redirectURL                 = "http://127.0.0.1/callback"
+	pinnipedCLIClientID         = oidcapi.ClientIDPinnipedCLI
+	contextTimeout              = 60 * time.Second
+	extraIDToken                = "id_token"
+	federationDomainPath        = "provider/pinniped"
+
+	mdKeyAuthToken    = "Authorization"
+	authTokenPrefix   = "Bearer "
+	mdKeyAuthIDToken  = "grpc-metadata-x-user-id"
+	mdKeyRefreshToken = "grpc-metadata-x-refresh-token"
+)
+
+type smSession struct {
+	sharedOauthConfig             *oauth2.Config
+	issuerURL, username, password string
+
```

Comment:
nit: new line. Post addressing this, the formatting asked above will work.

Reply:
Done